### PR TITLE
Estute/codejail ci

### PR DESCRIPTION
**Pull request by estute**
These are the necessary changes to get tests running on Jenkins, using workers built with this branch: https://github.com/edx/configuration/pull/5291

NOTES:
* Added a Jenkinsfile to run CI tests. For now, it only runs the `make test_no_proxy` target for python2.7, but we can expand the `parallel` block to include the other make target and additional Python environments.
* Added a tox file for testing other Python environments
* Needed to set the `CODEJAIL_TEST*` environment variables so that tests can see the sandboxed environment without changing how the code normally runs.
* Added the libraries needed in tests into a requirements file.

Additionally, I needed to fix some tests. Normally, I would let that be enough, but since this is somewhat more sensitive than a normal repo, I figured I would give some explanations:
* `TestLimits.test_cant_use_too_much_memory` was failing with the following error:
```
sudo: error in /etc/sudo.conf, line 0 while loading plugin `sudoers_policy'
sudo: unable to load /usr/lib/sudo/sudoers.so: /usr/lib/sudo/sudoers.so: failed to map segment from shared object
sudo: fatal error, unable to load plugins
```
This was because we were limiting the jailed code to `30000000`, but running Python via sudo hits a peak memory usage exceeding that for a second, prior to even executing the code meant to exercise the limit. I bumped the limit and code to exercise it as well.
* `TestLimits.test_cant_use_too_much_cpu` attempts to use more CPU time than it is allowed to, and is receiving a SIGXCPU, which should return a `24` but is getting a `-24`. I don't understand why.
* `TestLimits.test_cant_fork` was not actually blocking forking, so the test was failing. Limiting the jailed code to one process is blocking calls to fork.

### CODE REVIEW

**Comment by estute** [comment]
jenkins run

**Comment by estute** [comment]
jenkins run

**Comment by estute** [comment]
jenkins run

**Comment by estute** [comment]
jenkins run

**Comment by estute** [comment]
jenkins run

**Comment by estute** [comment]
jenkins run

**Comment by estute** [comment]
jenkins run

**Comment by estute** [comment]
jenkins run

File: codejail/tests/test_jail_code.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def test_directories_are_copied(self):
         self.assertResultOk(res)
         self.assertEqual(res.stdout, textwrap.dedent("""\
             ('.', ['pylib', 'tmp'], ['hello.txt', 'jailed_code'])
-            ('./pylib', [], ['module.py', 'module.pyc'])
+            ('./pylib', [], ['module.py'])
```
Comment (nedbat) [inline codejail/tests/test_jail_code.py]: I don't understand why the pyc is removed from the expected results?

Comment (estute) [reply codejail/tests/test_jail_code.py]: I don't know why we would expect it to be there. It's not checked into the repo (https://github.com/edx/codejail/tree/edx-platform-master/codejail/tests/pylib) and the `module.py` file isn't read by the interpreter, as it is copied to `/tmp` and the jailed code doesn't read it.

**Comment by estute** [comment]
jenkins run

**Comment by estute** [comment]
jenkins run

**Comment by estute** [comment]
jenkins run

**Comment by estute** [comment]
I found a flaky test `codejail.tests.test_jail_code.TestFeatures.test_argv `, hence the multiple `jenkins run` commands. I will clean that up in a separate PR

**Pull request merged**